src/cluster_legacy.c

```diff
@@ -4928,6 +4928,12 @@ int clusterDelSlot(int slot) {
 
     if (!n) return C_ERR;
 
+    /* Cleanup the empty dicts if myself is the owner. */
+    if (n == myself) {
```

Comment:
This seems a bit like it is breaking the cluster abstraction we implemented. I don't know why the underlying implementation would need to know what the format of the data is.